circle.yml

```diff
@@ -4,58 +4,114 @@
-  environment:
-    GOPATH: $HOME/go-path
-    PATH: $GOROOT/bin:$PATH
-    # Limit memory usage of gradle
-    _JAVA_OPTIONS: "-Xms512m -Xmx2048m"
-    GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError"'
-    REACT_NATIVE_MAX_WORKERS: 1
-    # Some node dependencies break without this
-    CXX: g++-4.8
-    ANDROID_NDK_HOME: ${ANDROID_NDK}
-general:
-  build_dir: .
-dependencies:
-  pre:
-    - sudo rm -rf $HOME/go* # Remove old go installation
-    - sudo rm -rf /usr/local/go* # Remove old go installation
-    - wget -nv https://storage.googleapis.com/golang/go1.9.linux-amd64.tar.gz -O $HOME/go.tar.gz
-    - mkdir $HOME/go-path
-    - (cd $HOME; tar -xzvf go.tar.gz)
-    - sudo mv $HOME/go /usr/local
-    - (cd $HOME; git clone https://github.com/keybase/client.git)
-    - mkdir -p $GOPATH/src/github.com/keybase
-    - ln -s $HOME/client $GOPATH/src/github.com/keybase/client
-    - if ! $(grep -q "Revision=24.4.1" $ANDROID_HOME/tools/source.properties); then echo y | android update sdk -u -a -t "tools"; fi
-    - if [ ! -e $ANDROID_HOME/build-tools/23.0.2 ]; then echo y | android update sdk -u -a -t "build-tools-23.0.2"; fi
-     # Android Support Repository, revision 35 / Local Maven repository for Support Libraries
-    - if [ ! -d "/usr/local/android-sdk-linux/extras/android/m2repository/com/android/support/design/24.1.0" ]; then echo y | android update sdk --no-ui --all --filter "extra-android-m2repository"; fi
+version: 2
+jobs:
+  build:
+    working_directory: ~/kbfs
+    parallelism: 1
+    shell: /bin/bash --login
+    environment:
+      GOVERSION: 1.10.3
+
+      # Some node dependencies break without this.
+      CXX: g++-4.8
+
+    # TODO: Consider picking a better docker image.
+    #
+    # For more information on choosing an image (or alternatively
+    # using a VM instead of a container) see
+    # https://circleci.com/docs/2.0/executor-types/ To see the list of
+    # pre-built images that CircleCI provides for most common
+    # languages see https://circleci.com/docs/2.0/circleci-images/
+    docker:
+    - image: circleci/build-image:ubuntu-14.04-XXL-upstart-1189-5614f37
+      command: /sbin/init
+
+    steps:
+    - run:
+        name: Update PATH and Define Environment Variable at Runtime
+        command: |
+          echo 'export GOPATH=$HOME/go-path' >> $BASH_ENV
+          echo 'export PATH=$HOME/.yarn/bin:$GOROOT/bin:$PATH' >> $BASH_ENV
+
+          # Some node dependencies break without this.
+          echo 'export ANDROID_NDK=$ANDROID_HOME/ndk-bundle' >> $BASH_ENV
+          echo 'export ANDROID_NDK_HOME=$ANDROID_HOME/ndk-bundle' >> $BASH_ENV
+
+          source $BASH_ENV
+
+    - checkout
+
+    # Install node and java.
+    - run: nvm install 10 && nvm alias default 10
+    - run: sudo update-alternatives --set java /usr/lib/jvm/jdk1.8.0/bin/java; sudo update-alternatives --set javac /usr/lib/jvm/jdk1.8.0/bin/javac; echo -e "export JAVA_HOME=/usr/lib/jvm/jdk1.8.0" >> $BASH_ENV
```

Comment:
Do they not give us a way to run these two steps as prerequisites? Also shouldn't these happen before checkout?